### PR TITLE
Refactor all API server calls from react to go through the services

### PR DESCRIPTION
**Pull request by ZmagoD**
https://biosistemika.atlassian.net/browse/SCI-1669
and
https://biosistemika.atlassian.net/browse/SCI-1727

### CODE REVIEW

File: app/javascript/src/services/api/endpoints.js [hunk 1]
```diff
@@ -8,17 +8,14 @@ export const ACTIVITIES_PATH = "/client_api/activities";
 
 // settings
 export const SETTINGS_PATH = "/settings";
```
Comment (Ducz0r) [inline app/javascript/src/services/api/endpoints.js]: Is this `SETTINGS_PATH` still needed here?

File: features/support/wait.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# @TODO make this funtion work!
```
Comment (Ducz0r) [inline features/support/wait.rb]: :banana:

File: flow-typed/npm/babel-plugin-transform-react-jsx-source_vx.x.x.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
+// flow-typed signature: 94a58ea1333bada1e79278664744bcec
```
Comment (Ducz0r) [inline flow-typed/npm/babel-plugin-transform-react-jsx-source_vx.x.x.js]: Should this file be in Git repo? Or should we ignore it?

File: flow-typed/npm/babel-polyfill_vx.x.x.js [hunk 4]
```diff
@@ -1,5 +1,5 @@
-// flow-typed signature: 7f2ed99d320afda9cbf973047b8b7ef6
```
Comment (Ducz0r) [inline flow-typed/npm/babel-polyfill_vx.x.x.js]: Should this file be in Git repo? Or should we ignore it?

File: flow-typed/teams_types.js [hunk 5]
```diff
@@ -21,12 +21,12 @@ export type Teams$NewTeam = {
...
 
-
 export type Teams$Team = {
   id: number,
   name: string,
   current_team: boolean,
   role: string,
   members: number,
-  can_be_leaved: boolean
+  can_be_leaved: boolean,
```
Comment (Ducz0r) [inline flow-typed/teams_types.js]: Sorry, grammar Nazi :). This should be written `can_be_left`. Can you rename this throughout the application?

File: flow-typed/npm/babel-plugin-transform-react-jsx-source_vx.x.x.js (hunk 3 above)
Comment (ZmagoD) [reply flow-typed/npm/babel-plugin-transform-react-jsx-source_vx.x.x.js]: I don't know this stuff is autogenerated I think that flow-typed fetch the types from npm and probably tracks this modules

Comment (Ducz0r) [reply flow-typed/npm/babel-plugin-transform-react-jsx-source_vx.x.x.js]: Yeah, I checked the [documentation](https://github.com/flowtype/flow-typed/wiki/FAQs#why-do-i-need-to-commit-the-libdefs-that-flow-typed-installs-for-my-project) and it should be included in Git. It's kind of additional overhead, but whatev.

**Pull request merged**